graphqlapi/src/main/java/io/stargate/graphql/schema/graphqlfirst/processor/DirectiveModelsBuilder.java

```diff
@@ -0,0 +1,296 @@
+            operationName, CqlDirectives.WHERE, inputValue.getName());
+        buildState.setFoundErrors();
+        return false;
+      }
+      if (!hasIfDirective && !hasCqlIncrementDirective) {
+        // Ignore non-annotated regular field (it's an update value, not a condition)
+        return false;
+      }
+      buildState.setWhereCondition(false);
+    }
+    return true;
+  }
+
+  private boolean validateIfHasErrors(
+      InputValueDefinition inputValue,
+      boolean hasWhereDirective,
+      boolean hasIfDirective,
+      boolean hasCqlIncrementDirective) {
+    if (hasWhereDirective && hasIfDirective) {
+      reportTwoAnnotationsSet(inputValue, CqlDirectives.WHERE, CqlDirectives.IF);
+      return true;
+    }
+
+    if (hasWhereDirective && hasCqlIncrementDirective) {
+      reportTwoAnnotationsSet(inputValue, CqlDirectives.WHERE, CqlDirectives.INCREMENT);
+      return true;
+    }
+
+    if (hasIfDirective && hasCqlIncrementDirective) {
+      reportTwoAnnotationsSet(inputValue, CqlDirectives.IF, CqlDirectives.INCREMENT);
+      return true;
+    }
+    return false;
+  }
+
+  private void reportTwoAnnotationsSet(
+      InputValueDefinition inputValue, String firstDirective, String secondDirective) {
+    invalidMapping(
+        "Operation %s: can't set both @%s and @%s on argument %s",
+        operationName, firstDirective, secondDirective, inputValue.getName());
+  }
+
+  private void reportInvalidMapping(InputValueDefinition inputValue, String directiveName) {
+    invalidMapping(
+        "Mutation %s: @%s is not allowed on primary key fields (%s)",
+        operationName, directiveName, inputValue.getName());
+  }
+
+  private Optional<FieldModel> findField(
+      InputValueDefinition inputValue, boolean hasIfDirective, boolean hasCqlIncrementDirective) {
+    String directiveName;
+    if (hasIfDirective) {
+      directiveName = CqlDirectives.IF;
+    } else if (hasCqlIncrementDirective) {
+      directiveName = CqlDirectives.INCREMENT;
+    } else {
+      directiveName = CqlDirectives.WHERE;
+    }
+    Optional<Directive> directive = DirectiveHelper.getDirective(directiveName, inputValue);
+    String fieldName =
+        directive
+            .flatMap(
+                d ->
+                    DirectiveHelper.getStringArgument(
+                        d, CqlDirectives.WHERE_OR_IF_OR_INCREMENT_FIELD, context))
+            .orElse(inputValue.getName());
+    Optional<FieldModel> result =
+        entity.getAllColumns().stream()
+            .filter(f -> f.getGraphqlName().equals(fieldName))
+            .findFirst();
+    if (!result.isPresent()) {
+      invalidMapping(
+          "Operation %s: could not find field %s in type %s",
+          operationName, fieldName, entity.getGraphqlName());
+    }
+    return result;
+  }
+
+  static class BuildState {
+
+    private boolean foundErrors = false;
+    private boolean isWhereCondition = false;
```

Comment:
Hrmm, not a big fan of this, although I understand that you did it to allow helper methods to modify the booleans.

The whole class is kind of hard to follow now. Maybe breaking down the three cases (select/delete/update) into submethods would help...